src/test/mir-opt/inline-closure-borrows-arg.rs

```diff
@@ -17,23 +17,48 @@ fn foo<T: Copy>(_t: T, q: &i32) -> i32 {
 
 // END RUST SOURCE
 // START rustc.foo.Inline.after.mir
-// ...
-// bb0: {
-//     ...
-//     _3 = [closure@HirId { owner: DefIndex(0:4), local_id: 31 }];
-//     ...
-//     _4 = &_3;
-//     ...
-//     _6 = &(*_2);
-//     ...
-//     _7 = &(*_2);
-//     _5 = (move _6, move _7);
-//     _8 = move (_5.0: &i32);
-//     _9 = move (_5.1: &i32);
-//     ...
-//     _0 = (*_8);
-//     ...
-//     return;
+// fn foo(_1: T, _2: &i32) -> i32{
+//     debug _t => _1;
+//     debug q => _2;
+//     let mut _0: i32;
+//     let mut _4: &[closure@HirId { owner: DefIndex(0:4), local_id: 31 }];
+//     let mut _5: (&i32, &i32);
+//     let mut _6: &i32;
+//     let mut _7: &i32;
+//     let mut _9: &i32;
+//     let mut _10: &i32;
+//     scope 1 {
+//         let _3: [closure@HirId { owner: DefIndex(0:4), local_id: 31 }];
+//     }
+//     scope 2 {
+//         debug x => _3;
+//         scope 3 {
+//             debug r => _9;
+//             debug _s => _10;
+//         }
+//     }
+//     scope 4 {
+//         let _8: &i32;
+//     }
+//     scope 5 {
+//         debug variable => _8;
```

Comment:
This is wrong, it should point to the same variable `debug r` does (thanks, @github, for eating my more detailed comment).
I think this might just be copyprop getting rid of the `variable = r` assignment without knowing about debuginfo.